src/openforms/js/components/admin/forms/MappingArrayInput.js

```diff
@@ -0,0 +1,153 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import DeleteIcon from 'components/admin/DeleteIcon';
+
+import ArrayInput from './ArrayInput';
+
+const MappingArrayInput = ({
+  name,
+  inputType,
+  mapping,
+  onChange,
+  deleteConfirmationMessage,
+  addButtonMessage,
+  valueArrayInput = false,
+}) => {
+  const intl = useIntl();
+
+  let copy = _.cloneDeep(mapping);
+
+  const onAdd = event => {
+    if (valueArrayInput) {
+      copy = copy.concat([['', ['']]]);
+    } else {
+      copy = copy.concat([['', '']]);
+    }
+
+    onChange(copy);
+  };
+
+  const onDelete = index => {
+    copy.splice(index, 1);
+
+    onChange(copy);
+  };
+
+  const onInputChange = (index, event) => {
+    if (event.target.name === 'keyInput') {
+      copy[index][0] = event.target.value;
+    } else if (event.target.name === 'valueInput') {
+      copy[index][1] = event.target.value;
+    }
+
+    onChange(copy);
+  };
+
+  const onArrayInputChange = (index, value) => {
+    copy[index][1] = value;
+    onChange(copy);
+  };
```

Comment:
maybe it'd be better to use `produce` from `immer` to deal with immutable/draft stores? Would be nice if we can at some point remove lodash too.